src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,203 @@
+Authors: Kalle Kytölä
+-/
+import measure_theory.measure.measure_space
+
+/-!
+# Weak convergence of (finite) measures
+
+This file will define the topology of weak convergence of finite measures and probability measures
+on topological spaces. The topology of weak convergence is the coarsest topology w.r.t. which
+for every bounded continuous `ℝ≥0`-valued function `f` on, the integration of `f` against the
+measure is continuous.
+
+TODOs:
+* Define the topologies (the current version only defines the types) via
+  `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`.
+* Prove that an equivalent definition of the topologies is obtained requiring continuity of
+  integration of bounded continuous `ℝ`-valued functions instead.
+* Include the portmanteau theorem on characterizations of weak convergence of (Borel) probability
+  measures.
+
+## Main definitions
+
+The main definitions are the types `finite_measure α` and `probability_measure α`.
+
+TODO:
+* Define the topologies on the above types.
+
+## Main results
+
+None yet.
+
+TODO:
+* Portmanteau theorem.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measure α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+The current first implementation of `finite_measure α` and `probability_measure α` is directly
+as subtypes of `measure α`, and the coercion to a function is the composition `ennreal.to_nnreal`
+and the coercion to function of `measure α`. Another alternative would be to use a bijection
+with `vector_measure α ℝ≥0` as an intermediate step. The choice of implementation should not have
+drastic downstream effects, so it can be changed later if appropriate.
+
+Potential advantages of using the `nnreal`-valued vector measure alternative:
+ * The coercion to function would avoid need to compose with `ennreal.to_nnreal`, the
+   `nnreal`-valued API could be more directly available.
+Potential drawbacks of the vector measure alternative:
+ * The coercion to function would lose monotonicity, as non-measurable sets would be defined to
+   have measure 0.
+ * No integration theory directly. E.g., the topology definition requires `lintegral` w.r.t.
+   a coercion to `measure α` in any case.
+
+## References
+
+* [Billingsley, *Convergence of probability measures*][billingsley1999]
+
+## Tags
+
+weak convergence of measures, finite measure, probability measure
+
+-/
+
+noncomputable theory
+open measure_theory
+open set
+open filter
+open_locale topological_space ennreal nnreal
+
+namespace measure_theory
+
+variables {α : Type*} [measurable_space α]
+
+/-- Finite measures are defined as the subtype of measures that have the property of being finite
+measures (i.e., their total mass is finite). -/
+abbreviation finite_measure (α : Type*) [measurable_space α] : Type* :=
+{ μ : measure α // is_finite_measure μ }
```

Comment:
```suggestion
{μ : measure α // is_finite_measure μ}
```
No space around braces when defining sets or types (contrary to braces used in proofs -- yes, our style rules are a little bit contrived :-)